airbyte-integrations/connectors/source-opsgenie/source_opsgenie/source.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from requests.exceptions import HTTPError
+from json.decoder import JSONDecodeError
+
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
+
+from .streams import Teams, Users, Alerts, UserTeams, Services, Incidents, Integrations
+
+# Source
+class SourceOpsgenie(AbstractSource):
+    @staticmethod
+    def get_authenticator(config: Mapping[str, Any]):
+        authenticator = TokenAuthenticator(config["api_token"], auth_method="GenieKey")
+        return authenticator
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+
+        try:
+            auth = self.get_authenticator(config)
+            api_endpoint = f"https://{config['endpoint']}/v2/account"
+
+            response = requests.get(
```

Comment:
@adamschmidt have you thought about any retry logic or rate limiting? There are some examples in other connectors you can borrow logic from. It's not a must for the alpha but will eventually become necessary for reliability.

Reply:
Considerations:
* Rate Limiting: https://docs.opsgenie.com/docs/api-rate-limiting
* Response Codes: https://docs.opsgenie.com/docs/response

Reply:
This is important for the streams, not so much the initial connection checks